CorpusCallosum/data/fsaverage_cc_template.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2025 AI in Medical Imaging, German Center for Neurodegenerative Diseases(DZNE), Bonn
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from pathlib import Path
+
+import nibabel as nib
+import numpy as np
+from scipy import ndimage
+
+from CorpusCallosum.shape.cc_postprocessing import process_slice
+
+
+def smooth_contour(contour, window_size=5):
+        """
+        Smooth a contour using a moving average filter.
+        
+        Parameters
+        ----------
+        contour : tuple of arrays
+            The contour coordinates (x, y).
+        window_size : int
+            Size of the smoothing window.
+            
+        Returns
+        -------
+        tuple of arrays
+            The smoothed contour coordinates (x, y).
+        """
+        x, y = contour
+        
+        # Ensure the window size is odd
+        if window_size % 2 == 0:
+            window_size += 1
+        
+        # Create a padded version of the arrays to handle the edges
+        x_padded = np.pad(x, (window_size//2, window_size//2), mode='wrap')
+        y_padded = np.pad(y, (window_size//2, window_size//2), mode='wrap')
+        
+        # Apply moving average
+        x_smoothed = np.zeros_like(x)
+        y_smoothed = np.zeros_like(y)
+        
+        for i in range(len(x)):
+            x_smoothed[i] = np.mean(x_padded[i:i+window_size])
+            y_smoothed[i] = np.mean(y_padded[i:i+window_size])
+        
+        return (x_smoothed, y_smoothed)
+
+def load_fsaverage_cc_template():
+    # smooth outside contour
+    # Apply smoothing to the outside contour using a moving average
+
+    try:
+        freesurfer_home = Path(os.environ['FREESURFER_HOME'])
+    except KeyError as err:
+        raise OSError(f"FREESURFER_HOME environment variable is not set correctly or does not exist: "
+                      f"{freesurfer_home}, either provide your own template or set the "
+                      f"FREESURFER_HOME environment variable") from err
+
+    fsaverage_seg_path = freesurfer_home / 'subjects' / 'fsaverage' / 'mri' / 'aparc+aseg.mgz'
+    fsaverage_seg = nib.load(fsaverage_seg_path)
+    segmentation = fsaverage_seg.get_fdata()
```

Comment:
it is highly recommended, that segmentations are not loaded with get_fdata().
This forces a conversion to float which is not good for segmentations
```suggestion
    segmentation = np.asarray(fsaverage_seg.dataobj)
```